eclair-core/src/main/scala/fr/acinq/eclair/Features.scala

```diff
@@ -154,4 +161,67 @@ object Features {
    */
   def areSupported(features: ByteVector): Boolean = areSupported(features.bits)
 
+  object Resolution {
+    private val knownFeatures: Seq[Feature] = Seq(
```

Comment:
How about using some reflection to guard us against forgetting new future features? How ugly would it look like?

Reply:
Not too ugly but we'd have to add a new dependency for `scala-reflect`:
```scala
    import scala.reflect.runtime.universe._
    import scala.reflect.runtime.{ universe => runtime }
    val mirror = runtime.runtimeMirror(ClassLoader.getSystemClassLoader)
    val subclasses = typeOf[Feature].typeSymbol.asClass.knownDirectSubclasses
    lazy val knownFeatures = subclasses.map({ desc =>
      val mod = mirror.staticModule(desc.asClass.fullName)
      mirror.reflectModule(mod).instance
    })
```

Reply:
As a compromise i added the test to guard us against forgetting new features but without using reflection at runtime. b2441ed